lib/Gocdb_Services/ServiceService.php

```diff
@@ -577,103 +577,153 @@ public function editService(\Service $se, $newValues, \User $user = null) {
+     *                ['hostname']    hostname of service being updated
+     *                ['description'] description of service being updated
+     *                ['url']         url of service being updated
+     *                ['dn']          dn of service being updated
+     *                ['$ip6']        ip V6 of service being updated
+     *                ['ip']          ip of service being updated
+     *                ['$os']         os of service being updated
+     *                ['email']       email of service being updated
+     *                ['$arch']       architecture  of service being updated
+     *                ['$monitored']  boolean monitored of service being updated
+     *                ['$beta']       boolean beta of service being updated
+     *                ['$production'] boolean production of service being updated
+     *                ['$notify']     boolean notify value of service being updated
+     * @throws Exception
+     */
+    private function editServiceLogic(\Service $service, $scopes, $sType, $updatedServiceValues){
+      // Explicity demarcate our tx boundary
+      $this->em->getConnection ()->beginTransaction ();
+      try {
+          // Set the service's member variables
+          $service->setHostName($updatedServiceValues['hostname']);
+          $service->setDescription($updatedServiceValues['description']);
+          $service->setUrl($updatedServiceValues['url']);
+          $service->setDn($updatedServiceValues['dn']);
+          $service->setIpAddress($updatedServiceValues['ip']);
+          $service->setIpV6Address($updatedServiceValues['ip6']);
+          $service->setOperatingSystem($updatedServiceValues['os']);
+          $service->setEmail($updatedServiceValues['email']);
+          $service->setArchitecture($updatedServiceValues['arch']);
+          $service->setMonitored($updatedServiceValues['monitored']);
+          $service->setBeta($updatedServiceValues['beta']);
+          $service->setProduction($updatedServiceValues['production']);
+          $service->setNotify($updatedServiceValues['notify']);
+
+          $service->setServiceType($sType);
+
+          // Update the service's scope
+          // firstly remove all existing scope links
+          $oldScopes = $service->getScopes ();
+          foreach($oldScopes as $s ) {
+              $service->removeScope($s);
+          }
 
-            // find then link each scope specified to the site
-            // foreach ($scopeIdsToApply as $scopeId) {
-            // $dql = "SELECT s FROM Scope s WHERE s.id = ?1";
-            // $scope = $this->em->createQuery($dql)
-            // ->setParameter(1, $scopeId)
-            // ->getSingleResult();
-            // $se->addScope($scope);
-            // }
-            foreach ( $selectedScopesToApply as $scope ) {
-                $se->addScope ( $scope );
-            }
+          // find then link each scope specified to the site
```

Comment:
I'm having trouble parsing this, is it:
"find then link each scope (specified to the site)"
or
"find then link (each scope specified) to the site"
?